pxtcompiler/emitter/service.ts

```diff
@@ -528,6 +530,16 @@ namespace ts.pxtc {
                                 source = foundSrc;
                             }
                             si.attributes = parseCommentString(source);
+
+                            if (existing.kind === SymbolKind.Module) {
+                                // Copy the existing package array to the new symbol info
+                                si.pkgs = existing.pkgs || []
+                                if (existing?.pkg !== si?.pkg) {
```

Comment:
you are passing a reference to the array. did you want to ``.slice(0)`` it?

Reply:
I actually do want it to be a reference vs copying it. We're replacing the `existing` symbol info in the `byQname` object with current symbol info `si` object, so the current symbol info will be the only one referencing the package array. Since the `existing` object reference will no won't come up again, there is no harm in copying it's reference to the package array vs copying the contents of said array.

I updated the comment to make my intention here clearer.